source/boundary_temperature/dynamic_core.cc

```diff
@@ -0,0 +1,993 @@
+            Tm1           =  prm.get_double ("Tm1");
+            Tm2           =  prm.get_double ("Tm2");
+            Theta         =  prm.get_double ("Theta");
+            composition_dependency
+              =  prm.get_bool("Composition dependency");
+            use_bw11      =  prm.get_bool("Use BW11");
+          }
+          prm.leave_subsection ();
+
+          prm.enter_subsection("Radioactive heat source");
+          {
+            n_radioheating_elements = prm.get_integer ("Number of radioactive heating elements");
+            heating_rate = Utilities::string_to_double
+                           (Utilities::split_string_list
+                            (prm.get("Heating rates")));
+            AssertThrow(n_radioheating_elements==heating_rate.size(),
+                        ExcMessage("Number of heating rate entities does not match "
+                                   "the number of radioactive elements."));
+            half_life = Utilities::string_to_double
+                        (Utilities::split_string_list
+                         (prm.get("Half life times")));
+            AssertThrow(n_radioheating_elements==half_life.size(),
+                        ExcMessage("Number of half life time entities does not match "
+                                   "the number of radioactive elements."));
+            initial_concentration = Utilities::string_to_double
+                                    (Utilities::split_string_list
+                                     (prm.get("Initial concentrations")));
+            AssertThrow(n_radioheating_elements==initial_concentration.size(),
+                        ExcMessage("Number of initial concentration entities does not match "
+                                   "the number of radioactive elements."));
+          }
+          prm.leave_subsection ();
+
+          prm.enter_subsection("Other energy source");
+          {
+            name_OES = prm.get("File name");
+          }
+          prm.leave_subsection ();
+
+          L=sqrt(3*K0*(log(Rho_cen/Rho_0)+1)/(2*M_PI*constants::big_g*Rho_0*Rho_cen));
+          D=sqrt(3*Cp/(2*M_PI*Alpha*Rho_cen*constants::big_g));
+
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+
+    template <int dim>
+    void
+    DynamicCore<dim>::read_data_OES()
+    {
+      data_OES.clear();
+      FILE *fp=fopen(name_OES.c_str(),"r");
+      if (fp!=NULL)
+        {
+          struct str_data_OES data_read;
+          while (!feof(fp))
+            {
+              if (fscanf(fp,"%le\t%le\n",&(data_read.t),&(data_read.w))==2)
+                data_OES.push_back(data_read);
+            }
+          fclose(fp);
+        }
+      if (data_OES.size()!=0)
+        this->get_pcout()<<"Other energy source is in use ( "<<data_OES.size()<<" data points is read)."<<std::endl;
+    }
+
+    template <int dim>
+    double
+    DynamicCore<dim>::get_OES(double t) const
+    {
+      t/=1.e9*year_in_seconds;
```

Comment:
Ah, I see. So if I understand you right, then you do this because all computations and constants for the core are usually done in units of billions of years because dynamics are just so slow? That makes sense. Would you mind adding a one-line comment to this effect?